openedx/tests/completion_aggregator_integration/test_creation.py

```diff
@@ -0,0 +1,323 @@
+                    parent_location=cls.sequential.location,
+                    category='vertical',
+                    modulestore=cls.store,
+                    publish_item=False,
+                )
+
+                cls.lc_block = ItemFactory.create(
+                    category="library_content",
+                    parent=cls.vertical,
+                    parent_location=cls.vertical.location,
+                    max_count=2,
+                    source_library_id=unicode(cls.library.location.library_key),
+                    modulestore=cls.store,
+                )
+                # copy children from library to content block (LibaryContentDescriptor.tools.update_children?)
+                cls.store.update_item(cls.course, creator.id)
+                cls.store.update_item(cls.lc_block, creator.id)
+
+    def setUp(self):
+        super(LibraryTestCase, self).setUp()
+        self.lc_block = self._refresh_children(self.lc_block)
+        self.in_course_blocks = [self.store.get_item(child) for child in self.lc_block.children]
+
+    def _refresh_children(self, lib_content_block):
+        """
+        Refresh the set of children on the library content block, then
+        fetch a new copy from the modulestore.
+        """
+        self.libtools.update_children(lib_content_block, self.user.id)
+        return self.store.get_item(lib_content_block.location)
+
+    @skip_unless_lms
+    def test_completing_library_content(self):
+        for block in self.in_course_blocks:
+            self.submit_completion_for(block, 0.75)
+        self.assert_expected_values({
+            self.vertical: (1.5, 2.0),
+            self.chapter: (1.5, 2.0),
+        })
+
+
+class EnrollmentTrackTestCase(_BaseTestCase):
+    """
+    Test that completion aggregates for blocks that are available to the user's
+    enrollment track only.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(EnrollmentTrackTestCase, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.course_key = cls.course.id
+        #import pdb; pdb.set_trace()
+        CourseMode.objects.create(course_id=cls.course_key, mode_slug=CourseMode.AUDIT, mode_display_name='Audit')
+        CourseMode.objects.create(
+            course_id=cls.course_key,
+            mode_slug=CourseMode.VERIFIED,
+            mode_display_name='Verified',
+            min_price=1.5
+        )
+        cls.partitions = partitions_service.get_all_partitions_for_course(cls.course)
+
+        with cls.store.bulk_operations(cls.course.id):
+            cls.chapter = ItemFactory.create(
+                parent=cls.course,
+                category="chapter",
+            )
+            cls.vertical1 = ItemFactory.create(
+                parent=cls.chapter,
+                category="vertical",
+                group_access={ENROLLMENT_TRACK_PARTITION_ID: [2]},  # 2 seems to be the value for verified track.  Where is this set?
```

Comment:
The `[2]` looks like it comes from your `setUpClass` above, where you create two `CourseMode` objects.  The first one is an `audit`, and the second is a `verified`.

Reply:
That makes some sense.  I'll try to confirm.